ql/src/test/queries/clientpositive/udf_trunc_number.q

```diff
@@ -71,4 +71,84 @@ DROP TABLE sampletable2;
 
 DROP TABLE sampletable3;
 
-DROP TABLE sampletable4;
\ No newline at end of file
+DROP TABLE sampletable4;
+
+select trunc(CAST('123456789.123456789' AS DECIMAL(18,9)), 8);
```

Comment:
This test seems to pass with or without the changes in this PR. Can you point which tests exactly fail due to overflow?